user deactivation email template

xcoulon opened the pull request:
Fixes #ODC345

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

alien-ike commented:
### Ike Plugins (test-keeper)

Thank you @xcoulon for this contribution!

<img align="left" src="https://raw.githubusercontent.com/arquillian/ike-prow-plugins/master/docs/images/arquillian_ui_failure_64px.png">

It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

xcoulon commented:
@corinnekrych @joshuawilson I added you both as reviewers for this PR since it contains an HTML email template to send to the user when her account is going to be deactivated. Would you mind taking a look at that file in particular? Also, I removed the `<div>` containing the OpenShift.io logo because the URL that we use is broken (`{{.webURL}}/_openshiftio/assets/images/openshift_logo_black.png`)

nurali-techie commented on README.adoc:
```diff
@@ -78,7 +78,7 @@ While running locally, you need to set `mandrill.apikey` configuration param.  T
 
-Run `make preview` to run https://github.com/fabric8-services/fabric8-notification/blob/master/preview/main.go[preview/main.go].  This will execute all template generation defined in main.go file.  This is useful when you want to check output of you template.  The output email will be stored at **./tmp/** folder with file name having Notification Type name as prefix.
+Run `make preview` to run https://github.com/fabric8-services/fabric8-notification/blob/master/preview/main.go[preview/main.go].  This will execute all template generation defined in main.go file.  This is useful when you want to check the output of your templatea.  The output email will be stored at **./tmp/** folder with file name having Notification Type name as prefix.
 
```
`templatea` typo.

xcoulon replied:
fixed in 9258bdb

xcoulon commented:
@nurali-techie @alexeykazakov @corinnekrych template is ready to review, thanks for @chritianvogt for his work 🙌

xcoulon commented:
FYI, running `make preview` produces the following output:
````
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <title>
      Notice: Your CodeReady Toolchain / OpenShift.io free account is inactive
    </title>
    <style>
      a:hover {
        text-decoration: underline !important;
      }
      p {
        text-align: left;
        margin: 30px 0;
      }
    </style>
  </head>

  <body
    style="
        padding: 10px;
        padding: 0;
        background-color: #f9f9f9;
        font-family: 'Open Sans', sans-serif;
        font-size: 15px;
        font-weight: lighter;
        line-height: 1.2;"
  >
    <div
      style="
        min-height: 300px;
        max-width: 750px;
        margin: 0 auto;
        padding: 20px;
        border: 1px solid #d7d7d7;
        border-radius: 4px;
        background-color: #fff;
        box-shadow: 0 2px 4px #d7d7d7;"
    >
      <img
        src="https://openshift.io/static/codeready_toolchain_dark.png"
        style="max-width: 75%"
        alt="CODEREADY TOOLCHAIN"
      />
      <div
        style="
            margin: 15px auto 0;
            border: 1px solid #d7d7d7;"
      ></div>

      <p>
        You are receiving this email because you have an OpenShift.io Online
        free account (user@example.com) and your action is required to
        keep it active.
      </p>

      <p>
        Login into the system by Apr 15, 2019 in order to continue
        using OpenShift.io uninterrupted.
      </p>

      <p>
        <a href="https://openshift.io">OpenShift.io</a> |
        <a href="https://che.openshift.io">Che.OpenShift.io</a>
      </p>

      <p>
        If you don't have time to use OpenShift Online right now, that's ok too.
        We will remove your access, delete the data and allocate resources to
        another user waiting to use the platform.
      </p>

      <p>
        Whenever you are ready, you can always recreate a development
        environment at <a href="https://openshift.io">OpenShift.io</a> to get
        access to the service again.
      </p>

      <p>
        Thanks,<br />
        The OpenShift.io Team
      </p>
    </div>
  </body>
</html>
````

nurali-techie commented on preview/main.go:
```diff
@@ -36,23 +37,22 @@ func main() {
+		{"51d968b1-b9e5-4ec1-884a-ff256902c753", string(types.CommentCreate)},
+		{"51d968b1-b9e5-4ec1-884a-ff256902c753", string(types.CommentUpdate)},
+		{"3383826c-51e4-401b-9ccd-b898f7e2397d", string(types.UserEmailUpdate)},
+		{"3383826c-51e4-401b-9ccd-b898f7e2397d", string(types.UserDeactivation)},
```
better to have diff uuid, plz change `3383826c-51e4-401b-9ccd-b898f7e2397d` to something else.

xcoulon replied:
ok

xcoulon replied:
done in 937aa01

nurali-techie commented:
@xcoulon plz add integration_test for `user.deactivation` type, similar to [this test](https://github.com/fabric8-services/fabric8-notification/blob/master/template/template_test.go#L231)

nurali-techie commented on template/user.deactivation/body.html:
```diff
@@ -0,0 +1,87 @@
+        If you don't have time to use OpenShift Online right now, that's ok too.
+        We will remove your access, delete the data and allocate resources to
+        another user waiting to use the platform.
+      </p>
```
In this, `If you don't have time to use OpenShift Online right now`
It should be "OpenShift.io Online" instead of "OpenShift Online" .. right ??

xcoulon replied:
the text comes from an external document, but I can ask for confirmation, indeed.

xcoulon replied:
actually, the term `OpenShift.io Online` is already used above, so I made the change you suggested in ff20314

nurali-techie commented on template/user.deactivation/body.html:
```diff
@@ -0,0 +1,87 @@
+      <p>
+        Login into the system by {{ .custom.expiryDate }} in order to continue
+        using OpenShift.io uninterrupted.
+      </p>
```
Will it help if we bold `expiryDate` to get the attention ??

xcoulon replied:
good point

xcoulon replied:
done in ff20314

nurali-techie replied:
@xcoulon I guess it's not `<bold>` but `<strong>` which will make text bold.

xcoulon replied:
ah right

xcoulon replied:
done in b4e8363

nurali-techie commented:
@xcoulon I guess you have missed one of the step while introducing new_type of notification. Plz have a look at readme doc [register](https://github.com/fabric8-services/fabric8-notification#register).  You need to add entry in [main.go](https://github.com/fabric8-services/fabric8-notification/blob/master/main.go#L104).

xcoulon commented:
> @xcoulon I guess you have missed one of the step while introducing new_type of notification. Plz have a look at readme doc [register](https://github.com/fabric8-services/fabric8-notification#register). You need to add entry in [main.go](https://github.com/fabric8-services/fabric8-notification/blob/master/main.go#L104).

@nurali-techie actually, I plan to address those changes in a subsequent PR, as part of https://jira.coreos.com/browse/ODC-219

nurali-techie commented:
> @nurali-techie actually, I plan to address those changes in a subsequent PR, as part of https://jira.coreos.com/browse/ODC-219

@xcoulon fine :+1:

nurali-techie commented:
@xcoulon I ran `make-preview` for user_deactivation and it gives one error in log.

```
ERRO[0010] unable to lookup user                         err="non 200 status code for GET user, returned 404" file=/home/nvirani/go/src/github.com/fabric8-services/fabric8-notification/collector/workitem.go func=github.com/fabric8-services/fabric8-notification/collector.resolveAllUsers line=255 pid=21923 pkg=github.com/fabric8-services/fabric8-notification/collector req_id= user_id=f3a184bf-963b-448a-86b4-307ae455223b
Subject: Notice: Your CodeReady Toolchain / OpenShift.io free account is inactive
Output : file:///home/nvirani/go/src/github.com/fabric8-services/fabric8-notification/tmp/user.deactivation-f3a184bf-963b-448a-86b4-307ae455223b.html
Headers:
```
The error is coming from [code](https://github.com/fabric8-services/fabric8-notification/blob/master/collector/workitem.go#L258)

Can you please check, I am not sure if this is due to the fact that `Resolver` part for this notification_type is not provided could be the root_cause of this.

xcoulon commented:
> code

Thanks for catching this. Fixed in df09c7f

alexeykazakov commented:
@nurali-techie you also could push to this PR branch instead of waiting for @xcoulon to address all the comment.

xcoulon commented:
@corinnekrych could you please give a review on this PR?

corinnekrych commented:
LGTM
